src/main/java/ejercicio/banco/repository/MySqlAccountRepository.java

```diff
@@ -2,22 +2,97 @@
 
 import ejercicio.banco.dto.Account;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.ArrayList;
 import java.util.List;
 
+import static ejercicio.banco.util.DataBaseUtil.closeConections;
+import static ejercicio.banco.util.DataBaseUtil.startConnection;
+
 public class MySqlAccountRepository implements AccountRepository {
+
+    private static final String FILENAME = "bank_project";
+
     @Override
     public List<Account> findAll(String filename) {
-        return null;
+        List<Account> accounts = new ArrayList();
+
+        Connection con = startConnection(filename);
+
+        try {
+            //Reads ACCOUNTS table, returning results
+            PreparedStatement pstmt = con.prepareStatement("select * from ACCOUNTS");
+            ResultSet rs = pstmt.executeQuery();
+            while (rs.next()) {
+                Integer id = rs.getInt("id");
+                String name = rs.getString("name");
+                Float money = rs.getFloat("money");
+                String iban = rs.getString("iban");
+                Account account = new Account(id, name, money, iban);
+                accounts.add(account);
+            }
+
+            closeConections(con);
```

Comment:
There is a potential issue here. Imagine your `while()`method throws an exception. If that ever happens then this line won't be executed (leaving an open connection with the database). Now imagine there are limited resources (available connections with the database). What would happen? 

When working with database connections (among other transactional communications), **every single connection must be closed regardless an exception happens or not**.

The way to achieve this is by using the try-catch-finally. Take a look to this article https://www.baeldung.com/java-exceptions (Section 4.3) and try to apply in the code.

Reply:
Is it necessary to use finally, or can I just use close after the try-catch (as in the last commit)?

Reply:
I would use `finally`simply because it is a good approach. Imagine that, in the future, you add a piece of code between the catch and the `closeConections()`, which throws another exception (and you don't know). You might be leaving that connection open.

In general, when using database connections, always:

```
Connection con;
try{
   // Code which throws WhateverException
} catch(WhateverException e) {
   // Manage exception somehow
} finally {
   if (con != null) {
      con.close(); // or however you close the connection
   }
}
```

Reply:
Ok